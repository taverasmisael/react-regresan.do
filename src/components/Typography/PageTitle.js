import React from 'react'
import PropTypes from 'prop-types'
import Typography from 'material-ui/Typography'

const PageTitle = ({ children, color = 'inherit', truncate = false, align = 'left' }) => (
  <Typography gutterBottom type="headline" color={color} align={align} noWrap={truncate}>
    {children}
  </Typography>
)

PageTitle.propTypes = {
  children: PropTypes.string,
  align: PropTypes.oneOf(['inherit', 'left', 'center', 'right', 'justify']),
  color: PropTypes.oneOf(['inherit', 'primary', 'secondary', 'accent', 'error', 'default']),
  truncate: PropTypes.bool
}

export default PageTitle
