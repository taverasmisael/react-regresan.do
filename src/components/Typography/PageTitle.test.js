import React from 'react'
import { shallow } from 'enzyme'

import PageTitle from './PageTitle'

describe('components/Typography/PageTitle', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<PageTitle />)
    })
    it('matches snapshot', () => {
      expect(<PageTitle />).toMatchSnapshot()
      expect(<PageTitle>Hello World</PageTitle>).toMatchSnapshot()
    })
  })

  describe('variants', () => {
    it('text display correctly', () => {
      const component = shallow(<PageTitle>Title</PageTitle>)
      expect(component.props().type).toBe('headline')
    })
  })
})
