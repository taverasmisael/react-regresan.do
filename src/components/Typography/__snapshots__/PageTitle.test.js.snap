// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography/PageTitle rendering matches snapshot 1`] = `<PageTitle />`;

exports[`components/Typography/PageTitle rendering matches snapshot 2`] = `
<PageTitle>
  Hello World
</PageTitle>
`;
